src/core/models/dt_linearmodel.cc

```diff
@@ -168,7 +174,8 @@ LinearModelFitOutput LinearModel<T>::fit_impl() {
 
         // Training
         dt::nested_for_static(iteration_size, ChunkSize(MIN_ROWS_PER_THREAD), [&](size_t i) {
-          size_t ii = (iteration_start + i) % dt_X_fit_->nrows();
+          // Do quasi-random data shuffling
+          size_t ii = ((iteration_start + i) * mp.multiplier + mp.increment) % dt_X_fit_->nrows();
```

Comment:
One problem with this approach is that it accesses data in semi-random pattern. In `kfold_random`, on the other hand, we randomly partition data into k parts such that each part's indices are sorted.